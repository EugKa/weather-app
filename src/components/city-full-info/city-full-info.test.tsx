import React from 'react';
import { shallow } from "enzyme";
import { CityFullInfo } from "..";

const props = {
   params: "London",
   current: {
    clouds: 40,
    dew_point: 9.31,
    dt: 1632751749,
    feels_like: 16.74,
    humidity: 59,
    pressure: 1013,
    sunrise: 1632722081,
    sunset: 1632764890,
    temp: 17.4,
    uvi: 1.65,
    visibility: 10000,
    weather: [
        {id: 802, main: 'Clouds', description: 'scattered clouds', icon: '03d'}
    ],
    wind_deg: 266,
    wind_gust: 3.58,
    wind_speed: 0.45,
   },
   timezone_offset : 3600
}


const setUp = (props:any) => shallow(<CityFullInfo {...props}/>)

describe('CityFullInfo component', () => {    
    it("should render CityFullInfo component with propsData", () => {
        const component = setUp(props)
        expect(component).toMatchSnapshot()
    }) 
})


