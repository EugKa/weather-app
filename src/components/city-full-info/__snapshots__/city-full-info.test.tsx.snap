// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityFullInfo component should render CityFullInfo component with propsData 1`] = `
<Styled(ForwardRef(Paper))>
  <ul
    className="todayList"
  >
    <li>
      Sep 27, 15:09pm
    </li>
    <h2>
      London
    </h2>
    <ul
      className="imgWrapper"
    >
      <ForwardRef(CardMedia)
        component="img"
        height={70}
        image="http://openweathermap.org/img/wn/03d@2x.png"
      />
      <li>
        17
        ℃
      </li>
    </ul>
    <ul
      className="info"
    >
      <li>
        <span
          className="infoText"
        >
          Feels like 
          16
          ℃.
        </span>
        <span
          className="infoText"
        >
          Scattered clouds
          .
        </span>
      </li>
    </ul>
    <ul
      className="info"
    >
      <li>
        <span
          className="infoText"
        >
          Wind speed: 
          0.45
        </span>
        <span
          className="infoText"
        >
          Preshure: 
          1013
          hPa
        </span>
      </li>
      <li>
        <span
          className="infoText"
        >
          Humidity: 
          59
          %
        </span>
        <span
          className="infoText"
        >
          UV: 
          1.65
        </span>
      </li>
      <li>
        <span
          className="infoText"
        >
          Dew point: 
          9.31
          ℃
        </span>
        <span
          className="infoText"
        >
          Visibility: 
          10km
        </span>
      </li>
    </ul>
  </ul>
</Styled(ForwardRef(Paper))>
`;
